smurfs/src/reducers/index.js

```diff
@@ -1,23 +1,55 @@
-/*
-  Be sure to import in all of the action types from `../actions`
-*/
+import {
+  FETCH_SMURF,
+  CREATE_SMURF,
+  ADDING_SMURF_SUCC,
+  ADDING_SMURF_FAIL,
+  DELETING_SMURF
+} from '../actions';
 
-/*
- Your initial/default state for this project could *Although does not have to* look a lot like this
- {
-   smurfs: [],
-   fetchingSmurfs: false
-   addingSmurf: false
-   updatingSmurf: false
-   deletingSmurf: false
-   error: null
- }
-*/
+const initialState = {
+    smurfs: [],
+    fetchingSmurfs: false,
+    addingSmurf: false,
+    updatingSmurf: false,
+    deletingSmurf: false,
+    error: null
+};
 
-/*
-  You'll only need one smurf reducer for this project.
-  Feel free to export it as a default and import as rootReducer. 
-  This will guard your namespacing issues.
-  There is no need for 'combineReducers' in this project.
-  Components can then read your store as, `state` and not `state.fooReducer`.
-*/
+function reducer( state = initialState, action) {
+  switch (action.type) {
+    case FETCH_SMURF:
+      return {
+        ...state,
+        error: '',
+        addingSmurf: false
+      };
+    case CREATE_SMURF:
+      return {
+        ...state,
+        error: '',
+        addingSmurf: true
+      }
+    case ADDING_SMURF_SUCC:
+      return {
+        ...state,
+        error: '',
```

Comment:
two errors defined here...one is null and one is an empty string. You also did not switch your addingSmurf flag back to false...You might want to do that since on success, it's done adding.